finder-app/writer.c

```diff
@@ -0,0 +1,61 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <syslog.h>
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+
+int main ( int argc, char *argv[])
+{
+	openlog(NULL, 0, LOG_USER);
+
+	if (argc != 3)
+	{
+		printf("Invalid arguments. Please enter correct arguments \n\r");
+		printf("Expected arguments as <file_name> <string_to_write> \n\r");
+		syslog(LOG_ERR, "Invalid number of arguments: %d", argc);
+		return 1;
+	}
+
+	int fd;
+	size_t write_bytes= strlen(argv[2]);
+	ssize_t write_return;
+
+	fd=open (argv[1], O_WRONLY | O_CREAT | O_TRUNC, S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
+
+	if (fd==-1)
+	{
+		syslog(LOG_ERR, "File %s does not exist. Please create file first", argv[1]);
+		return 1;
+	}
+	
+	write_return = write(fd, argv[2], strlen(argv[2]));
```

Comment:
@anuh7 suggest using a variable here rather than referencing `argv` in multiple places.